lib/trivial.nix

```
@@ -446,7 +446,7 @@ rec {
      annotated with function args.
   */
   isFunction = f: builtins.isFunction f ||
-    (f ? __functor && isFunction (f.__functor f));
+    (f ? __functor && lib.trivial.isFunction (f.__functor f));
```

```suggestion
    (f ? __functor && isFunction (f.__functor f));
```

Performance regression. Use `let inherit (lib) isFunction;` near start of file.

---

Please do the same for the rest - at least for everything that doesn't perform I/O like `warn`.

---

Done.

Btw I'm curious, is "dereferencing" an attribute-set really that more expensive? (not _that_ deep into the evaluator I must admit :sweat_smile: )

---

No, it'd only be measurable if it's in a hot path, but functions like `isFunction` probably are. Not worth risking.
Also less noise.